src/AzureOpenAIProxy.Management/Services/ModelService.cs

```diff
@@ -1,25 +1,62 @@
+using System.Data;
 using AzureOpenAIProxy.Management.Components.ModelManagement;
 using AzureOpenAIProxy.Management.Database;
 using Microsoft.EntityFrameworkCore;
+using Npgsql;
+using NpgsqlTypes;
 
 namespace AzureOpenAIProxy.Management.Services;
 
-public class ModelService(IAuthService authService, AoaiProxyContext db) : IModelService
+public class ModelService(IAuthService authService, AoaiProxyContext db, IConfiguration configuration) : IModelService
 {
+
+    private const string PostgresEncryptionKey = "PostgresEncryptionKey";
+    private readonly NpgsqlConnection connection = (NpgsqlConnection)db.Database.GetDbConnection();
+
+    private async Task<byte[]?> PostgresEncryptValue(string value)
+    {
+        if (connection.State != ConnectionState.Open)
+            await connection.OpenAsync();
+
+        string? postgresEncryptionKey = configuration[PostgresEncryptionKey];
+
+        using var command = new NpgsqlCommand($"SELECT aoai.pgp_sym_encrypt('{value}', '{postgresEncryptionKey}');", connection);
```

Comment:
This isn't correct, we shouldn't use string interpolation, we should use parameterised SQL, as this is how we'd end up wit ha SQL injection attack